# Beacon SES data from the Dungog East Coast Low

This notebook imports and explores the cleaned up data from the Beacon SES call outs for the Dungog East Coast Low event. The data have been cleaned to fix errors in lat/lon.

The working directory for this project is: 

\\nas\gemd\georisk\HaRIA_B_Wind\projects\impact_forecasting\data\exposure\NSW\NSW_SES

*Date: September 2017, Creator: Claire Krause*

In [1]:
%matplotlib inline
import matplotlib.pyplot as plt

import pandas as pd
import numpy as np

import seaborn as sns
sns.set_context("poster")

# Start by reading in an exploring the dataset

In [64]:
filename = "//nas/gemd/georisk/HaRIA_B_Wind/projects/impact_forecasting/data/exposure/NSW/NSW_SES/Beacon_data_cleaned.csv"
df = pd.read_csv(filename)
df

Identifier  Latitude  Longitude  \
0      0002-6338   -33.409    151.402   
1      0002-6339   -33.669    151.095   
2      0002-6342   -33.061    151.577   
3      0002-6343   -33.097    151.631   
4      0002-6344   -33.790    151.194   
5      0002-6345   -32.191    152.518   
6      0002-6348   -32.174    152.498   
7      0002-6351   -33.759    151.075   
8      0002-6352   -33.492    151.314   
9      0002-6357   -33.717    151.116   
10     0002-6363   -33.804    151.289   
11     0002-6365   -33.751    151.241   
12     0002-6369   -32.740    151.879   
13     0002-6373   -33.021    151.679   
14     0002-6376   -32.928    151.754   
15     0002-6377   -33.446    151.438   
16     0002-6386   -32.960    151.685   
17     0002-6389   -32.759    151.780   
18     0002-6390       NaN        NaN   
19     0002-6396   -33.650    151.320   
20     0002-6403   -32.772    151.635   
21     0002-6404   -32.889    151.737   
22     0002-6405   -33.750    151.106   
23     0002-6408   -33.708    151.138   
24     0002-6410   -33.701    151.100   
25     0002-6412       NaN        NaN   
26     0002-6413       NaN        NaN   
27     0002-6419   -33.798    151.284   
28     0002-6426   -33.741    151.072   
29     0002-6429   -33.617    151.143   
...          ...       ...        ...   
13766  0004-6513   -33.799    151.082   
13767  0004-6514   -32.670    152.113   
13768  0004-6515   -33.392    151.351   
13769  0004-6520   -33.792    151.266   
13770  0004-6523   -33.033    151.607   
13771  0004-6525   -33.487    151.429   
13772  0004-6526       NaN        NaN   
13773  0004-6528   -33.258    151.514   
13774  0004-6529   -33.443    151.327   
13775  0004-6530   -32.890    151.586   
13776  0004-6533   -33.428    151.386   
13777  0004-6534   -32.933    151.595   
13778  0004-6537   -32.987    151.609   
13779  0004-6538   -32.731    151.877   
13780  0004-6539   -32.888    151.717   
13781  0004-6540   -33.263    151.330   
13782  0004-6541   -33.800    151.129   
13783  0004-6543   -32.990    151.656   
13784  0004-6544   -32.993    151.673   
13785  0004-6545   -32.757    151.758   
13786  0004-6546   -32.993    151.673   
13787  0004-6547   -32.812    151.482   
13788  0004-6551   -31.193    152.823   
13789  0004-7308   -32.812    151.653   
13790  0004-7310   -32.810    151.656   
13791  0004-8482   -33.546    151.012   
13792  0005-0483   -32.468    151.962   
13793  0005-1376   -34.099    150.702   
13794  0005-1377       NaN        NaN   
13795  0005-2189   -31.549    159.077   

                                             JobReceived  Priority  \
0      Mon Apr 20 2015 06:27:51 GMT+1000 (AUS Eastern...   General   
1      Mon Apr 20 2015 07:24:48 GMT+1000 (AUS Eastern...   General   
2      Mon Apr 20 2015 08:09:38 GMT+1000 (AUS Eastern...   General   
3      Mon Apr 20 2015 08:45:21 GMT+1000 (AUS Eastern...   General   
4      Mon Apr 20 2015 08:58:03 GMT+1000 (AUS Eastern...  Priority   
5      Mon Apr 20 2015 09:11:25 GMT+1000 (AUS Eastern...   General   
6      Mon Apr 20 2015 09:16:29 GMT+1000 (AUS Eastern...   General   
7      Mon Apr 20 2015 09:38:36 GMT+1000 (AUS Eastern...   General   
8      Mon Apr 20 2015 09:55:33 GMT+1000 (AUS Eastern...   General   
9      Mon Apr 20 2015 10:21:49 GMT+1000 (AUS Eastern...   General   
10     Mon Apr 20 2015 10:50:27 GMT+1000 (AUS Eastern...   General   
11     Mon Apr 20 2015 11:05:17 GMT+1000 (AUS Eastern...   General   
12     Mon Apr 20 2015 11:15:23 GMT+1000 (AUS Eastern...   General   
13     Mon Apr 20 2015 11:43:58 GMT+1000 (AUS Eastern...   General   
14     Mon Apr 20 2015 12:04:54 GMT+1000 (AUS Eastern...   General   
15     Mon Apr 20 2015 12:16:45 GMT+1000 (AUS Eastern...   General   
16     Mon Apr 20 2015 12:34:22 GMT+1000 (AUS Eastern...   General   
17     Mon Apr 20 2015 12:41:39 GMT+1000 (AUS Eastern...   General   
18     Mon Apr 20 2015 12:42:14 GMT+1000 (AUS Eastern...   General   
19     Mon Apr 20 2015 12:55:23 GMT+1000 (AUS Eastern...   Gene

### List of column headings

In [65]:
df.columns

Index([u'Identifier', u'Latitude', u'Longitude', u'JobReceived', u'Priority',
       u'Type', u'SituationOnScene', u'Headquarters', u'Region',
       u'Address.Street', u'Address.Locality', u'Address.PostCode', u'Tags',
       u'Tags.1', u'Tags.2', u'Tags.3', u'Tags.4', u'Tags.5', u'Tags.6',
       u'Tags.7', u'Tags.8', u'Tags.9', u'Tags.10', u'Tags.11', u'Tags.12',
       u'Tags.13', u'Tags.14'],
      dtype='object')

### How many entries do we have for each column?
I.e. not NaN. Note that this does not check the contents of the field, only that something was written into it

In [66]:
df.count()

Identifier          13796
Latitude            13491
Longitude           13491
JobReceived         13796
Priority            13796
Type                13796
SituationOnScene    12060
Headquarters        13796
Region              13796
Address.Street      13340
Address.Locality    13363
Address.PostCode    11215
Tags                 4487
Tags.1               3987
Tags.2               3187
Tags.3               2498
Tags.4               1884
Tags.5               1320
Tags.6                875
Tags.7                579
Tags.8                359
Tags.9                204
Tags.10                98
Tags.11                47
Tags.12                20
Tags.13                 6
Tags.14                 2
dtype: int64

# Beacon data for the Hunter Region ECL event:

* Number of data entries = 13796
* Lat/lon = 13491 (97.8%)
* SituationOnScene (free text description of call-out) = 12060 (87.4%)
* Tags (at least one) = 4487 (32.5%)

Key missing data:
* Damage state
* Property type and attributes

# Just do some quick cleaning up of the data

There are a number of blank fields throughout the data where a value was not entered into the dataset by the assessor. We need to keep track of the missing data, as well as the entered data, so we will find all 'NaN' values in the dataset, and change these to 'Not given' so we can include them in subsequent analyses. 

In [67]:
df = df.fillna('Not given')

# The tags field looks pretty valuable. We will separate the tags into their own columns so we can explore the frequency of different tags

In [69]:
grouped = Counter(df[['Tags', 'Tags.1', 'Tags.2', 'Tags.3', 'Tags.4', 'Tags.5', 'Tags.6', 'Tags.7', 'Tags.8', 
                      'Tags.9', 'Tags.10', 'Tags.11', 'Tags.12', 'Tags.13', 'Tags.14']].values.flatten())
grouped

Counter({'1 Storey': 1054,
         '2 Storeys': 518,
         '> 2 Storeys': 86,
         'Aged': 339,
         'Aged Care': 17,
         'Air Recon': 7,
         'Air Support': 4,
         'Animal': 1,
         'Animals': 657,
         'Assist Evac': 1,
         'Av Slope Roof': 460,
         'Basement': 6,
         'Bedroom': 141,
         'Blocked Drain': 10,
         'Branch Down': 387,
         'Branch Threatening': 301,
         'Bridge': 6,
         'Building': 784,
         'Building Collapse': 26,
         'Burst Gas': 3,
         'Burst Sewer': 7,
         'Burst Water': 6,
         'Car / Vehicle': 106,
         'Caravan / Mobile Home': 12,
         'Carer': 22,
         'Catering Transport': 1,
         'Cattle': 3,
         'Causeway': 2,
         'Ceiling Damage': 177,
         'Chain Sawing': 175,
         'Clean-up Assistance': 1,
         'Colourbond': 200,
         'Commercial': 43,
         'Council Land': 57,
         'Council/Government': 71,
         'Crown Land'

## Grab any call outs relating to downed trees
3337 entries for tree damage

In [79]:
tree_damage = df.loc[df.isin(['Branch Down', 'Branch Threatening', 'Tree Down', 'Tree Threatening']).any(1)]
tree_damage.shape

(3337, 27)

## Grab any call outs relating to severe building damage
72 entries for severe building damage

In [80]:
severe_damage = df.loc[df.isin(['Building Collapse', 'Threat of Collapse']).any(1)]
severe_damage.shape

(72, 27)

## Grab any call outs relating to roof damage
1231 entries for roof damage

In [82]:
roof_damage = df.loc[df.isin(['Ceiling Damage', 'Leaking Roof', 'Roof Damage', 'Tarping']).any(1)]
roof_damage.shape

(1231, 27)